### ISSUE TITLE
syntastic enabled for second file in diff mode (vimdiff)

### ISSUE THREAD

**Issue by hvhaugwitz** [scrooloose/syntastic#1643]
When opening two files in diff mode (vimdiff) syntastic is enabled for the second file though it should be disabled (according to #125):
Example:
```
$ printf '#!/bin/sh\nexit 0' > /tmp/a.sh
$ printf '#!/bin/sh\nUNUSED=\nexit 0' > /tmp/b.sh
$ vimdiff /tmp/a.sh /tmp/b.sh
$ vimdiff /tmp/b.sh /tmp/a.sh
```
With shellcheck installed the first call of vimdiff shows the shellcheck warning while the second call is clean.

**Comment by lcd047** [comment]
Yes, syntastic + active mode + `check_on_open` + opening files in multiple windows = trouble.  To the best of my knowledge this can't be solved without crippling useful functionality (see #1184 and #400 for related fun).  You can avoid the problem by turning off `check_on_open`, or by making the relevant filetypes passive.  But between the quirks of Vim startup sequence and the limitations of Vim's API, this can't be solved in a robust way.  There are many [other corner cases](https://github.com/scrooloose/syntastic/labels/known%20bugs) that fail for similar reasons.  Syntastic is just a best effort approach, most of the time it works by accident rather than by design. _shrug_

**Comment by lcd047** [comment]
Hopefully fixed by f527cb1.

**Comment by hvhaugwitz** [comment]
Yes, the issue is fixed. Many thanks.
